File: clients/csharp/src/main/csharp/io/swagger/Api/MoviesApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using RestSharp;
using IO.Swagger.Client;
using IO.Swagger.Model;

namespace IO.Swagger.Api {
  

  public interface IMoviesApi {
    
    /// <summary>
    /// Returns all movies 
    /// </summary>
    
    /// <returns>List<Movie></returns>
    List<Movie> GetMovieList ();

    /// <summary>
    /// Returns all movies 
    /// </summary>
    
    /// <returns>List<Movie></returns>
    Task<List<Movie>> GetMovieListAsync ();
    
    /// <summary>
    /// Create a new movie. Cannot reuse an existing id.
    /// </summary>
    /// <param name="Id">Movie&#39;s id</param>/// <param name="Title">Title of the movie</param>/// <param name="Director">Who directed the movie</param>/// <param name="Synopsis">Short description about the movie</param>/// <param name="YearOfRelease">Year it was released</param>
    /// <returns></returns>
    void Create (int? Id, string Title, string Director, string Synopsis, string YearOfRelease);

    /// <summary>
    /// Create a new movie. Cannot reuse an existing id.
    /// </summary>
    /// <param name="Id">Movie&#39;s id</param>/// <param name="Title">Title of the movie</param>/// <param name="Director">Who directed the movie</param>/// <param name="Synopsis">Short description about the movie</param>/// <param name="YearOfRelease">Year it was released</param>
    /// <returns></returns>
    Task CreateAsync (int? Id, string Title, string Director, string Synopsis, string YearOfRelease);
    
    /// <summary>
    /// Returns a single movie for an id You have to supply a valid id.
    /// </summary>
    /// <param name="Id">Id of the movie to return</param>
    /// <returns>Movie</returns>
    Movie GetMovieById (int? Id);

    /// <summary>
    /// Returns a single movie for an id You have to supply a valid id.
    /// </summary>
    /// <param name="Id">Id of the movie to return</param>
    /// <returns>Movie</returns>
    Task<Movie> GetMovieByIdAsync (int? Id);
    
  }

  /// <summary>
  /// Represents a collection of functions to interact with the API endpoints
  /// </summary>
  public class MoviesApi : IMoviesApi {

    /// <summary>
    /// Initializes a new instance of the <see cref="MoviesApi"/> class.
    /// </summary>
    /// <param name="apiClient"> an instance of ApiClient (optional)
    /// <returns></returns>
    public MoviesApi(ApiClient apiClient = null) {
      if (apiClient == null) { // use the default one in Configuration
        this.apiClient = Configuration.apiClient; 
      } else {
        this.apiClient = apiClient;
      }
    }

    /// <summary>
    /// Initializes a new instance of the <see cref="MoviesApi"/> class.
    /// </summary>
    /// <returns></returns>
    public MoviesApi(String basePath)
    {
      this.apiClient = new ApiClient(basePath);
    }

    /// <summary>
    /// Sets the base path of the API client.
    /// </summary>
    /// <value>The base path</value>
    public void SetBasePath(String basePath) {
      this.apiClient.basePath = basePath;
    }

    /// <summary>
    /// Gets the base path of the API client.
    /// </summary>
    /// <value>The base path</value>
    public String GetBasePath(String basePath) {
      return this.apiClient.basePath;
    }

    /// <summary>
    /// Gets or sets the API client.
    /// </summary>
    /// <value>The API client</value>
    public ApiClient apiClient {get; set;}


    
    /// <summary>
    /// Returns all movies 
    /// </summary>
    
    /// <returns>List<Movie></returns>
    public List<Movie> GetMovieList () {

      

      var path = "/movies";
      path = path.Replace("{format}", "json");
      

      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, String>();
      var fileParams = new Dictionary<String, String>();
      String postBody = null;

      
      
      
      

      // authentication setting, if any
      String[] authSettings = new String[] {  };

      // make the HTTP request
      IRestResponse response = (IRestResponse) apiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);

      if (((int)response.StatusCode) >= 400) {
        throw new ApiException ((int)response.StatusCode, "Error calling GetMovieList: " + response.Content, response.Content);
      }
      return (List<Movie>) apiClient.Deserialize(response.Content, typeof(List<Movie>));
    }
	
	 /// <summary>
    /// Returns all movies 
    /// </summary>
    
    /// <returns>List<Movie></returns>
    public async Task<List<Movie>> GetMovieListAsync () {

      

      var path = "/movies";
      path = path.Replace("{format}", "json");
      

      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, String>();
      var fileParams = new Dictionary<String, String>();
      String postBody = null;

      
      
      
      

      // authentication setting, if any
      String[] authSettings = new String[] {  };

      // make the HTTP request
      IRestResponse response = (IRestResponse) await apiClient.CallApiAsync(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
      if (((int)response.StatusCode) >= 400) {
        throw new ApiException ((int)response.StatusCode, "Error calling GetMovieList: " + response.Content, response.Content);
      }
      return (List<Movie>) apiClient.Deserialize(response.Content, typeof(List<Movie>));
    }
    
    /// <summary>
    /// Create a new movie. Cannot reuse an existing id.
    /// </summary>
    /// <param name="Id">Movie&#39;s id</param>/// <param name="Title">Title of the movie</param>/// <param name="Director">Who directed the movie</param>/// <param name="Synopsis">Short description about the movie</param>/// <param name="YearOfRelease">Year it was released</param>
    /// <returns></returns>
    public void Create (int? Id, string Title, string Director, string Synopsis, string YearOfRelease) {

      

      var path = "/movies/create";
      path = path.Replace("{format}", "json");
      

      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, String>();
      var fileParams = new Dictionary<String, String>();
      String postBody = null;

      
      
      if (Id != null) formParams.Add("id", apiClient.ParameterToString(Id)); // form parameter
      if (Title != null) formParams.Add("title", apiClient.ParameterToString(Title)); // form parameter
      if (Director != null) formParams.Add("director", apiClient.ParameterToString(Director)); // form parameter
      if (Synopsis != null) formParams.Add("synopsis", apiClient.ParameterToString(Synopsis)); // form parameter
      if (YearOfRelease != null) formParams.Add("yearOfRelease", apiClient.ParameterToString(YearOfRelease)); // form parameter
      
      

      // authentication setting, if any
      String[] authSettings = new String[] {  };

      // make the HTTP request
      IRestResponse response = (IRestResponse) apiClient.CallApi(path, Method.POST, queryParams, postBody, headerParams, formParams, fileParams, authSettings);

      if (((int)response.StatusCode) >= 400) {
        throw new ApiException ((int)response.StatusCode, "Error calling Create: " + response.Content, response.Content);
      }
      
      return;
    }
	
	 /// <summary>
    /// Create a new movie. Cannot reuse an existing id.
    /// </summary>
    /// <param name="Id">Movie&#39;s id</param>/// <param name="Title">Title of the movie</param>/// <param name="Director">Who directed the movie</param>/// <param name="Synopsis">Short description about the movie</param>/// <param name="YearOfRelease">Year it was released</param>
    /// <returns></returns>
    public async Task CreateAsync (int? Id, string Title, string Director, string Synopsis, string YearOfRelease) {

      

      var path = "/movies/create";
      path = path.Replace("{format}", "json");
      

      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, String>();
      var fileParams = new Dictionary<String, String>();
      String postBody = null;

      
      
      if (Id != null) formParams.Add("id", apiClient.ParameterToString(Id)); // form parameter
      if (Title != null) formParams.Add("title", apiClient.ParameterToString(Title)); // form parameter
      if (Director != null) formParams.Add("director", apiClient.ParameterToString(Director)); // form parameter
      if (Synopsis != null) formParams.Add("synopsis", apiClient.ParameterToString(Synopsis)); // form parameter
      if (YearOfRelease != null) formParams.Add("yearOfRelease", apiClient.ParameterToString(YearOfRelease)); // form parameter
      
      

      // authentication setting, if any
      String[] authSettings = new String[] {  };

      // make the HTTP request
      IRestResponse response = (IRestResponse) await apiClient.CallApiAsync(path, Method.POST, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
      if (((int)response.StatusCode) >= 400) {
        throw new ApiException ((int)response.StatusCode, "Error calling Create: " + response.Content, response.Content);
      }
      
      return;
    }
    
    /// <summary>
    /// Returns a single movie for an id You have to supply a valid id.
    /// </summary>
    /// <param name="Id">Id of the movie to return</param>
    /// <returns>Movie</returns>
    public Movie GetMovieById (int? Id) {

      
      // verify the required parameter 'Id' is set
      if (Id == null) throw new ApiException(400, "Missing required parameter 'Id' when calling GetMovieById");
      

      var path = "/movies/{id}";
      path = path.Replace("{format}", "json");
      path = path.Replace("{" + "id" + "}", apiClient.ParameterToString(Id));
      

      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, String>();
      var fileParams = new Dictionary<String, String>();
      String postBody = null;

      
      
      
      

      // authentication setting, if any
      String[] authSettings = new String[] {  };

      // make the HTTP request
      IRestResponse response = (IRestResponse) apiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);

      if (((int)response.StatusCode) >= 400) {
        throw new ApiException ((int)response.StatusCode, "Error calling GetMovieById: " + response.Content, response.Content);
      }
      return (Movie) apiClient.Deserialize(response.Content, typeof(Movie));
    }
	
	 /// <summary>
    /// Returns a single movie for an id You have to supply a valid id.
    /// </summary>
    /// <param name="Id">Id of the movie to return</param>
    /// <returns>Movie</returns>
    public async Task<Movie> GetMovieByIdAsync (int? Id) {

      
          // verify the required parameter 'Id' is set
          if (Id == null) throw new ApiException(400, "Missing required parameter 'Id' when calling GetMovieById");
      

      var path = "/movies/{id}";
      path = path.Replace("{format}", "json");
      path = path.Replace("{" + "id" + "}", apiClient.ParameterToString(Id));
      

      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, String>();
      var fileParams = new Dictionary<String, String>();
      String postBody = null;

      
      
      
      

      // authentication setting, if any
      String[] authSettings = new String[] {  };

      // make the HTTP request
      IRestResponse response = (IRestResponse) await apiClient.CallApiAsync(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
      if (((int)response.StatusCode) >= 400) {
        throw new ApiException ((int)response.StatusCode, "Error calling GetMovieById: " + response.Content, response.Content);
      }
      return (Movie) apiClient.Deserialize(response.Content, typeof(Movie));
    }
    
  }  
  
}

File: clients/java/src/main/java/io/swagger/client/api/MoviesApi.java
package io.swagger.client.api;

import io.swagger.client.ApiException;
import io.swagger.client.ApiClient;
import io.swagger.client.Configuration;

import io.swagger.client.model.*;

import java.util.*;

import io.swagger.client.model.Movie;

import com.sun.jersey.multipart.FormDataMultiPart;
import com.sun.jersey.multipart.file.FileDataBodyPart;

import javax.ws.rs.core.MediaType;

import java.io.File;
import java.util.Map;
import java.util.HashMap;

public class MoviesApi {
  private ApiClient apiClient;

  public MoviesApi() {
    this(Configuration.getDefaultApiClient());
  }

  public MoviesApi(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  public ApiClient getApiClient() {
    return apiClient;
  }

  public void setApiClient(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  
  /**
   * Returns all movies
   * 
   * @return List<Movie>
   */
  public List<Movie> getMovieList () throws ApiException {
    Object postBody = null;
    

    // create path and map variables
    String path = "/movies".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, String> formParams = new HashMap<String, String>();

    

    

    final String[] accepts = {
      "application/json"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    if(contentType.startsWith("multipart/form-data")) {
      boolean hasFields = false;
      FormDataMultiPart mp = new FormDataMultiPart();
      
      if(hasFields)
        postBody = mp;
    }
    else {
      
    }

    try {
      String[] authNames = new String[] {  };
      String response = apiClient.invokeAPI(path, "GET", queryParams, postBody, headerParams, formParams, accept, contentType, authNames);
      if(response != null){
        return (List<Movie>) apiClient.deserialize(response, "array", Movie.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
  /**
   * Create a new movie.
   * Cannot reuse an existing id.
   * @param id Movie&#39;s id
   * @param title Title of the movie
   * @param director Who directed the movie
   * @param synopsis Short description about the movie
   * @param yearOfRelease Year it was released
   * @return void
   */
  public void create (Integer id, String title, String director, String synopsis, String yearOfRelease) throws ApiException {
    Object postBody = null;
    

    // create path and map variables
    String path = "/movies/create".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, String> formParams = new HashMap<String, String>();

    

    

    final String[] accepts = {
      "text/plain"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    if(contentType.startsWith("multipart/form-data")) {
      boolean hasFields = false;
      FormDataMultiPart mp = new FormDataMultiPart();
      
      if (id != null) {
        hasFields = true;
        mp.field("id", apiClient.parameterToString(id), MediaType.MULTIPART_FORM_DATA_TYPE);
      }
      
      if (title != null) {
        hasFields = true;
        mp.field("title", apiClient.parameterToString(title), MediaType.MULTIPART_FORM_DATA_TYPE);
      }
      
      if (director != null) {
        hasFields = true;
        mp.field("director", apiClient.parameterToString(director), MediaType.MULTIPART_FORM_DATA_TYPE);
      }
      
      if (synopsis != null) {
        hasFields = true;
        mp.field("synopsis", apiClient.parameterToString(synopsis), MediaType.MULTIPART_FORM_DATA_TYPE);
      }
      
      if (yearOfRelease != null) {
        hasFields = true;
        mp.field("yearOfRelease", apiClient.parameterToString(yearOfRelease), MediaType.MULTIPART_FORM_DATA_TYPE);
      }
      
      if(hasFields)
        postBody = mp;
    }
    else {
      if (id != null)
        formParams.put("id", apiClient.parameterToString(id));
      if (title != null)
        formParams.put("title", apiClient.parameterToString(title));
      if (director != null)
        formParams.put("director", apiClient.parameterToString(director));
      if (synopsis != null)
        formParams.put("synopsis", apiClient.parameterToString(synopsis));
      if (yearOfRelease != null)
        formParams.put("yearOfRelease", apiClient.parameterToString(yearOfRelease));
      
    }

    try {
      String[] authNames = new String[] {  };
      String response = apiClient.invokeAPI(path, "POST", queryParams, postBody, headerParams, formParams, accept, contentType, authNames);
      if(response != null){
        return ;
      }
      else {
        return ;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
  /**
   * Returns a single movie for an id
   * You have to supply a valid id.
   * @param id Id of the movie to return
   * @return Movie
   */
  public Movie getMovieById (Integer id) throws ApiException {
    Object postBody = null;
    
    // verify the required parameter 'id' is set
    if (id == null) {
       throw new ApiException(400, "Missing the required parameter 'id' when calling getMovieById");
    }
    

    // create path and map variables
    String path = "/movies/{id}".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "id" + "\\}", apiClient.escapeString(id.toString()));

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, String> formParams = new HashMap<String, String>();

    

    

    final String[] accepts = {
      "application/json"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    if(contentType.startsWith("multipart/form-data")) {
      boolean hasFields = false;
      FormDataMultiPart mp = new FormDataMultiPart();
      
      if(hasFields)
        postBody = mp;
    }
    else {
      
    }

    try {
      String[] authNames = new String[] {  };
      String response = apiClient.invokeAPI(path, "GET", queryParams, postBody, headerParams, formParams, accept, contentType, authNames);
      if(response != null){
        return (Movie) apiClient.deserialize(response, "", Movie.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
}
